src/builtin.c

```diff
@@ -757,9 +757,8 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     input = f_tostring(jq, input);
     const unsigned char* data = (const unsigned char*)jv_string_value(input);
     int len = jv_string_length_bytes(jv_copy(input));
-    size_t decoded_len = (3 * (size_t)len) / 4; // 3 usable bytes for every 4 bytes of input
+    size_t decoded_len = (3 * (size_t)len) / 4 + 1; // 3 usable bytes for every 4 bytes of input
```

Comment:
Alternative i guess could be to do a early return if input is an empty string?

Reply:
Fixed.

Reply:
No! The fix was not correct. I should have remembered why I did +1 here.
```
 $./jq -n '"a" | @base64d'
Assertion failed: (nemb > 0 && sz > 0), function jv_mem_calloc, file jv_alloc.c, line 154.
zsh: abort      ./jq -n '"a" | @base64d'
```